perception/autoware_traffic_light_arbiter/test/test_node.cpp

```diff
@@ -0,0 +1,276 @@
+}
+
+void generatePerceptionMsg(TrafficSignalArray & perception_msg)
+{
+  // traffic_light_group_id 1: 1012
+  {
+    TrafficSignal traffic_light_groups;
+    traffic_light_groups.traffic_light_group_id = 1012;
+    // elements 1: red + circle
+    {
+      TrafficElement elements;
+      elements.color = TrafficElement::RED;
+      elements.shape = TrafficElement::CIRCLE;
+      elements.status = TrafficElement::SOLID_ON;
+      elements.confidence = 0.9;
+      traffic_light_groups.elements.push_back(elements);
+    }
+    // elements 2: green + right arrow
+    {
+      TrafficElement elements;
+      elements.color = TrafficElement::GREEN;
+      elements.shape = TrafficElement::RIGHT_ARROW;
+      elements.status = TrafficElement::SOLID_ON;
+      elements.confidence = 0.8;
+      traffic_light_groups.elements.push_back(elements);
+    }
+    perception_msg.traffic_light_groups.push_back(traffic_light_groups);
+  }
+}
+
+void generateExternalMsg(TrafficSignalArray & external_msg)
+{
+  // traffic_light_group_id 1: 1012
+  {
+    TrafficSignal traffic_light_groups;
+    traffic_light_groups.traffic_light_group_id = 1012;
+    // elements 1: green + circle
+    {
+      TrafficElement elements;
+      elements.color = TrafficElement::GREEN;
+      elements.shape = TrafficElement::CIRCLE;
+      elements.status = TrafficElement::SOLID_ON;
+      elements.confidence = 0.6;
+      traffic_light_groups.elements.push_back(elements);
+    }
+    // elements 2: green + right arrow
+    {
+      TrafficElement elements;
+      elements.color = TrafficElement::GREEN;
+      elements.shape = TrafficElement::RIGHT_ARROW;
+      elements.status = TrafficElement::SOLID_ON;
+      elements.confidence = 0.5;
+      traffic_light_groups.elements.push_back(elements);
+    }
+    external_msg.traffic_light_groups.push_back(traffic_light_groups);
+  }
+}
+
+bool isMsgEqual(const TrafficSignalArray & input_msg, const TrafficSignalArray & gt_msg)
+{
+  // check number of groups
+  if (input_msg.traffic_light_groups.size() != gt_msg.traffic_light_groups.size()) {
+    return false;
+  }
+
+  for (size_t group_idx = 0; group_idx < input_msg.traffic_light_groups.size(); ++group_idx) {
+    const auto & input_traffic_light_group = input_msg.traffic_light_groups.at(group_idx);
+    const auto & gt_traffic_light_group = gt_msg.traffic_light_groups.at(group_idx);
+
+    // check traffic_light_group_id
+    if (
+      input_traffic_light_group.traffic_light_group_id !=
+      gt_traffic_light_group.traffic_light_group_id) {
+      return false;
+    }
+
+    // check number of elements
+    if (input_traffic_light_group.elements.size() != gt_traffic_light_group.elements.size()) {
+      std::cout << input_traffic_light_group.elements.size() << std::endl;
+      std::cout << gt_traffic_light_group.elements.size() << std::endl;
+      return false;
+    }
+
+    for (size_t element_idx = 0; element_idx < input_traffic_light_group.elements.size();
```

Comment:
```suggestion
    for (std::size_t element_idx = 0; element_idx < input_traffic_light_group.elements.size();
```

Reply:
Thank you for your review. I fixed in https://github.com/autowarefoundation/autoware.universe/pull/8747/commits/58c928b160251fd8a374964eeb2ee3eeba9002ac